theano/gpuarray/linalg.py

```diff
@@ -385,25 +396,19 @@ def make_node(self, A):
         assert A.dtype == 'float32'
         if self.compute_uv:
             return theano.Apply(self, [A],
-                                [A.type(),
-                                GpuArrayType(A.dtype, broadcastable=[False],
-                                             context_name=ctx_name)(),
-                                A.type()])
+                                # return S, U, VT
+                                [GpuArrayType(A.dtype, broadcastable=[False],
+                                              context_name=ctx_name)(),
```

Comment:
this change the interface. We should add a detection somewhre that if an old pickled file reload the old order, then it raise an error when w try to make a thunk of this. I think you should  implement the check in th emethod prepare_node()